batch-oai-harvest/src/main/java/data/unit/reader/OaiRepositoryIdentifiersHarvester.java

```diff
@@ -0,0 +1,89 @@
+package data.unit.reader;
+
+import data.entity.ExecutionRecordExternalIdentifier;
+import data.entity.ExecutionRecordKey;
+import eu.europeana.metis.harvesting.HarvesterException;
+import eu.europeana.metis.harvesting.HarvesterFactory;
+import eu.europeana.metis.harvesting.ReportingIteration;
+import eu.europeana.metis.harvesting.oaipmh.OaiHarvest;
+import eu.europeana.metis.harvesting.oaipmh.OaiHarvester;
+import eu.europeana.metis.harvesting.oaipmh.OaiRecordHeader;
+import eu.europeana.metis.harvesting.oaipmh.OaiRecordHeaderIterator;
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.LinkedList;
+import java.util.List;
+
+@Component
+@StepScope
+public class OaiRepositoryIdentifiersHarvester implements ItemReader<ExecutionRecordExternalIdentifier> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @Value("#{jobParameters['oaiEndpoint']}")
+    private String oaiEndpoint;
+    @Value("#{jobParameters['oaiSet']}")
+    private String oaiSet;
+    @Value("#{jobParameters['oaiMetadataPrefix']}")
+    private String oaiMetadataPrefix;
+    @Value("#{jobParameters['datasetId']}")
+    private String datasetId;
+    @Value("#{stepExecution.jobExecution.jobInstance.id}")
+    private Long jobInstanceId;
+
+    final OaiHarvester oaiHarvester = HarvesterFactory.createOaiHarvester();
+    private final List<OaiRecordHeader> oaiRecordHeaders = new LinkedList<>();
```

Comment:
We should work on updating the OaiRecordHeaderIterator so that we have a method which we can call iteratively and in parallel and not just a `foreach` method. That will help us to make this parallelized.

For harvesting I see in the configuration that it is in fact not using parallelization and it's the reader anyway.